quarkc/test/e2e/primitives.q

```diff
@@ -304,6 +304,87 @@ class StringTest {
         assertEqual(false, "abc" == "def");
     }
 
+    void upper_lower() {
```

Comment:
Rather than growing primitives.q even more, can we put these type-specific tests into separate quark files? Maybe something like `bi_string.q` (for built-in) or `lib_list.q` or whatever? Something consistent would be great.